oblib/validator.py

```diff
@@ -77,12 +77,29 @@ def xbrli_boolean_item_type_validator(value, concept):
     """XBRLI boolean validator."""
     errors = []
     if type(value) is str:
-        if value.lower() not in ['true', 'false']:
+        if re.search(r'\d', value):
+            if len(re.findall(r'0*(1|0)', value)) is not 1:
+                errors += ["'{}' is not a valid boolean value.".format(value)]
+        elif value.lower() not in ['true', 'false']:
+            errors += ["'{}' is not a valid boolean value.".format(value)]
+    elif type(value) is int:
+        if not (value is 1 or value is 0):
             errors += ["'{}' is not a valid boolean value.".format(value)]
     elif type(value) is not bool:
         errors += ["'{}' is not a valid boolean value.".format(value)]
     return errors
 
+def xbrli_decimal_item_type_validator(value, concept):
+    """XBRLI decimal validator"""
+    errors = []
+    if type(value) is str:
+        if re.search(r'[^0-9.]', value):
+            errors += ["'{} is not a valid decimal value.".format(value)]
```

Comment:
Use `errors.append("'{} is not a valid decimal value.".format(value))` instead of `+= [...]`